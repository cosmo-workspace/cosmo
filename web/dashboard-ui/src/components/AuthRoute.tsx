import React, { ReactElement } from 'react';
import { Navigate, useLocation } from 'react-router-dom';
import { isAdminUser } from '../views/organisms/UserModule';
import { useLogin } from './LoginProvider';

type Props = {
  children: ReactElement;
  admin?: boolean;
}

export const AuthRoute: React.VFC<Props> = ({ children, admin }) => {
  const { loginUser } = useLogin();
  const isAdmin = isAdminUser(loginUser);
  let location = useLocation();

  if (!loginUser) {
    return (<Navigate
      to={{ pathname: '/signin', search: `from=${location.pathname}` }}
      state={{ from: location }}
      replace
    />);
  } else if (admin && !isAdmin) {
    return (<Navigate to='/' replace />);
  } else {
    return children;
  }
}