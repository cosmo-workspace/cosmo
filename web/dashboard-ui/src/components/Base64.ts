// https://github.com/google/webauthndemo/blob/main/src/public/scripts/base64url.ts

const chars =
  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";

// Use a lookup table to find the index.
const lookup = new Uint8Array(256);
for (let i = 0; i < chars.length; i++) {
  lookup[chars.charCodeAt(i)] = i;
}

const encode = function (arraybuffer: ArrayBuffer): string {
  const bytes = new Uint8Array(arraybuffer);
  const len = bytes.length;
  let base64 = "";

  for (let i = 0; i < len; i += 3) {
    base64 += chars[bytes[i] >> 2];
    base64 += chars[((bytes[i] & 3) << 4) | (bytes[i + 1] >> 4)];
    base64 += chars[((bytes[i + 1] & 15) << 2) | (bytes[i + 2] >> 6)];
    base64 += chars[bytes[i + 2] & 63];
  }

  if (len % 3 === 2) {
    base64 = base64.substring(0, base64.length - 1);
  } else if (len % 3 === 1) {
    base64 = base64.substring(0, base64.length - 2);
  }

  return base64;
};

const decode = function (base64: string): ArrayBuffer {
  const len = base64.length;
  const bufferLength = base64.length * 0.75;
  const arraybuffer = new ArrayBuffer(bufferLength);
  const bytes = new Uint8Array(arraybuffer);

  let p = 0;
  for (let i = 0; i < len; i += 4) {
    const encoded1 = lookup[base64.charCodeAt(i)];
    const encoded2 = lookup[base64.charCodeAt(i + 1)];
    const encoded3 = lookup[base64.charCodeAt(i + 2)];
    const encoded4 = lookup[base64.charCodeAt(i + 3)];

    bytes[p++] = (encoded1 << 2) | (encoded2 >> 4);
    bytes[p++] = ((encoded2 & 15) << 4) | (encoded3 >> 2);
    bytes[p++] = ((encoded3 & 3) << 6) | (encoded4 & 63);
  }

  return arraybuffer;
};

const base64url = { encode, decode };
export { base64url };
