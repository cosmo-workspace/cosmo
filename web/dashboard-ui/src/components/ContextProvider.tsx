import { createContext, ReactNode, useContext, useState } from "react";

/**
 * DialogContext
 */
type DialogState<T> = { open: boolean, dialogProps?: T }
type DialogProps<T> = T & { onClose: () => void }
type Dispatch<T> = (open: boolean, dialogProp?: T) => void;

export function DialogContext<T>(dialog: (props: DialogProps<T>) => any) {

  const Context = createContext<{ state: DialogState<T>, dispatch: Dispatch<T> }>(undefined as any);

  return {
    Provider: ({ children }: { children: ReactNode }) => {
      const [state, setState] = useState<DialogState<T>>({ open: false });

      const dispatch: Dispatch<T> = (open, dialogProps?) => {
        setState({ open, dialogProps });
      }
      const closeHandler = () => dispatch(false, state.dialogProps);
      return (
        <Context.Provider value={{ state, dispatch }}><>
          {children}
          {state.open && dialog({ ...state.dialogProps, onClose: () => closeHandler() } as DialogProps<T>)}
        </></Context.Provider>
      );
    },
    useDispatch: () => useContext(Context).dispatch,
  }
}

/**
 * ModuleContext
 */
export function ModuleContext<T extends () => any>(useModule: T) {
  const Context = createContext<ReturnType<T>>(undefined as any);
  return {
    Provider: ({ children }: { children: ReactNode }) => {
      const module = useModule();
      return (<Context.Provider value={{ ...module }}>{children}</Context.Provider>);
    },
    useContext: () => useContext(Context),
  }
}
