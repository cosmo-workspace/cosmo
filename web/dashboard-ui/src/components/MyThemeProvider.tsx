import { colors, useMediaQuery } from "@mui/material";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import React, { useMemo } from "react";

const MyTheme = () => {
  console.log('MyTheme');
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)', { noSsr: true });

  return useMemo(() => {

    return createTheme({
      components: {
        MuiOutlinedInput: {
          defaultProps: {
            notched: true,
          },
        },
        MuiInputLabel: {
          defaultProps: {
            shrink: true,
          },
        },
      },
      palette: {
        mode: prefersDarkMode ? 'dark' : undefined,
        primary: colors.deepPurple,
        secondary: colors.pink,
      },
    });
  }, [prefersDarkMode]);
}

export const MyThemeProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {

  const myTheme = MyTheme();

  return (
    <ThemeProvider theme={myTheme}>
      {children}
    </ThemeProvider>
  );
};
