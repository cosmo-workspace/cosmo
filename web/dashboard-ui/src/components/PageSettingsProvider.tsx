import React, { createContext, useContext, useEffect, useReducer } from 'react';

interface PageSettings {
  isOpen: boolean;
};

interface PageSettingsContext {
  pageSettings: PageSettings;
  setPageSettings: (setting: PageSettings) => void;
  updatePageSettings: (setting: PageSettings) => void;
};

const Context = createContext<PageSettingsContext>({
  pageSettings: {
    isOpen: true,
  },
  setPageSettings: (_setting) => { },
  updatePageSettings: (_setting) => { },
});

interface Action {
  type: 'SET' | 'UPDATE';
  pageSettings: PageSettings;
}
function reducer(state: PageSettings, action: Action) {
  switch (action.type) {
    case 'SET':
      return action.pageSettings;
    case 'UPDATE':
      return { ...state, ...action.pageSettings }
    default:
      throw new Error()
  }
}

export const PageSettingsProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
  const persistKey = 'csm_PageSettings';
  const persistPageSettings = JSON.parse(localStorage.getItem('csm_PageSettings') || "{}");
  const [pageSettings, dispatch] = useReducer(reducer, persistPageSettings || {})

  useEffect(() => {
    try {
      localStorage.setItem(persistKey, JSON.stringify(pageSettings))
    } catch (error) {
      console.warn(error)
    }
  }, [pageSettings])

  return (
    <Context.Provider value={{
      pageSettings,
      setPageSettings: (settings) => { dispatch({ type: 'SET', pageSettings: settings }) },
      updatePageSettings: (settings) => { dispatch({ type: 'UPDATE', pageSettings: settings }) }
    }}>
      {children}
    </Context.Provider>
  )
}

export function usePageSettings() {
  return useContext(Context);
}
