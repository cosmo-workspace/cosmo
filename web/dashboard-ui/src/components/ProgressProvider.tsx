import { Backdrop, CircularProgress } from '@mui/material';
import React, { createContext, useContext, useState } from 'react';

const DispatchContext = createContext<React.Dispatch<React.SetStateAction<number>>>(undefined as any);

/**
 * provider
 */
export const ProgressProvider: React.FC = ({ children }) => {
  const [count, setCount] = useState(0);
  return (<>
    <DispatchContext.Provider value={setCount}>
      {children}
    </DispatchContext.Provider>
    <Backdrop sx={{ zIndex: (theme) => theme.zIndex.drawer + 1000 }} open={count > 0}>
      <CircularProgress />
    </Backdrop>
  </>);
}

export function useProgress() {
  const setCount = useContext(DispatchContext);
  return {
    setMask: () => setCount(count => { console.log('setMask', count + 1); return count + 1 }),
    releaseMask: () => setCount(count => { console.log('releaseMask', count); return count > 0 ? count - 1 : 0 }),
  }
}