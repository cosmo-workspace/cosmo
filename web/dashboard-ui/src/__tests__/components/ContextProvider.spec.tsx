import { Button, Dialog } from "@mui/material";
import { act, cleanup, render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import React, { useState } from "react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import { DialogContext, ModuleContext } from "../../components/ContextProvider";

//--------------------------------------------------
// test
//--------------------------------------------------
describe("DialogContext", () => {

  beforeEach(async () => {
  });

  afterEach(() => {
    vi.restoreAllMocks();
    cleanup();
  });

  const TestDialog: React.VFC<{ onClose: () => void, hoge: string }> = ({ onClose, hoge }) => {
    return (
      <Dialog open={true} onClose={() => onClose()}>
      </Dialog>
    )
  }

  const testDialogContext = DialogContext<{ hoge: string }>(
    props => (<TestDialog {...props} />));

  it("open/close", async () => {

    const Stub = () => {
      const dispatch = testDialogContext.useDispatch();
      return (<>
        <Button onClick={() => dispatch(true)}>open</Button>
        <Button onClick={() => dispatch(false)}>close</Button>
      </>);
    }
    const user = userEvent.setup();
    render(<testDialogContext.Provider><Stub /></testDialogContext.Provider>);
    await act(async () => { expect(document.body).toMatchSnapshot('1.initial render'); });
    await act(async () => { await user.click(screen.getByText('open')); });
    await act(async () => { expect(document.body).toMatchSnapshot('2.opend'); });
    await act(async () => { await user.click(screen.getByText('close')); });
    await act(async () => { expect(document.body).toMatchSnapshot('3.closed'); });
  });

  it("close esc", async () => {

    const Stub = () => {
      const dispatch = testDialogContext.useDispatch();
      return (<>
        <Button onClick={() => dispatch(true)}>open</Button>
        <Button onClick={() => dispatch(false)}>close</Button>
      </>);
    }
    const user = userEvent.setup();
    render(<testDialogContext.Provider><Stub /></testDialogContext.Provider>);
    await act(async () => { await user.click(screen.getByText('open')); });
    await act(async () => { await user.keyboard('{Esc}'); });
    await act(async () => { expect(document.body).toMatchSnapshot('closed'); });
  });

});


describe('ModuleContext', () => {

  it('normal', async () => {

    const useTestHook = () => {
      const [state, setState] = useState("aaaa");
      return { state, setState }
    }

    const TestContext = ModuleContext(useTestHook);
    const useTest = TestContext.useContext;

    const Component1: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
      const { state, setState } = useTest();
      return (<div>
        <button onClick={() => { setState("11111") }}>button1</button>
        <div data-testid="textbox1">{state}</div>
      </div>);
    }
    const Component2: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
      const { state, setState } = useTest();
      return (<div>
        <button onClick={() => { setState("22222") }}>button2</button>
        <div data-testid="textbox2">{state}</div>
      </div>);
    }
    const user = userEvent.setup();
    render(
      <TestContext.Provider>
        <Component1 />
        <Component2 />
      </TestContext.Provider>
    );

    await user.click(screen.getByText('button1'));
    expect(screen.getByTestId('textbox1')).toHaveTextContent('11111');
    expect(screen.getByTestId('textbox2')).toHaveTextContent('11111');
    await user.click(screen.getByText('button2'));
    expect(screen.getByTestId('textbox1')).toHaveTextContent('22222');
    expect(screen.getByTestId('textbox2')).toHaveTextContent('22222');
  });

});
