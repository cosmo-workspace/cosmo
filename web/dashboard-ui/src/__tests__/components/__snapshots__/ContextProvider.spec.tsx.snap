// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogContext > close esc > closed 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(0);
    -moz-transform: scale(0);
    -ms-transform: scale(0);
    transform: scale(0);
    opacity: 0.1;
  }

  100% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
    opacity: 0.3;
  }
}

@keyframes animation-1 {
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
}

@keyframes animation-2 {
  0% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    -webkit-transform: scale(0.92);
    -moz-transform: scale(0.92);
    -ms-transform: scale(0.92);
    transform: scale(0.92);
  }

  100% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-2 {
  opacity: 0;
  position: absolute;
}

.emotion-2.MuiTouchRipple-rippleVisible {
  opacity: 0.3;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-duration: 550ms;
  animation-duration: 550ms;
  -webkit-animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-2.MuiTouchRipple-ripplePulsate {
  -webkit-animation-duration: 200ms;
  animation-duration: 200ms;
}

.emotion-2 .MuiTouchRipple-child {
  opacity: 1;
  display: block;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: currentColor;
}

.emotion-2 .MuiTouchRipple-childLeaving {
  opacity: 0;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  -webkit-animation-duration: 550ms;
  animation-duration: 550ms;
  -webkit-animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-2 .MuiTouchRipple-childPulsate {
  position: absolute;
  left: 0px;
  top: 0;
  -webkit-animation-name: animation-2;
  animation-name: animation-2;
  -webkit-animation-duration: 2500ms;
  animation-duration: 2500ms;
  -webkit-animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-delay: 200ms;
  animation-delay: 200ms;
}

<body
  style=""
>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-focusVisible MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root emotion-1"
      >
        <span
          class="emotion-2 MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
          style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
          />
        </span>
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
  </div>
</body>
`;

exports[`DialogContext > open/close > 1.initial render 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
  </div>
</body>
`;

exports[`DialogContext > open/close > 2.opend 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-4 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

@media print {
  .emotion-4 {
    position: absolute!important;
  }
}

.emotion-5 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-6 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-6 {
    height: auto;
  }
}

.emotion-7 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);
  margin: 32px;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: calc(100% - 64px);
  max-width: 600px;
}

@media print {
  .emotion-7 {
    overflow-y: visible;
    box-shadow: none;
  }
}

@media (max-width:663.95px) {
  .emotion-7.MuiDialog-paperScrollBody {
    max-width: calc(100% - 64px);
  }
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
  </div>
  <div
    class="MuiDialog-root MuiModal-root emotion-4"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop emotion-5"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-6"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm emotion-7"
        role="dialog"
      />
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`DialogContext > open/close > 3.closed 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body
  style=""
>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root emotion-1"
      />
    </button>
  </div>
</body>
`;
